core/src/main/java/io/questdb/cutlass/line/array/AbstractArray.java

```diff
@@ -94,6 +134,61 @@ public void close() {
         closed = true;
     }
 
+    /**
+     * Reshapes the array to the specified dimensions.
+     * <p>This method allows changing the array's shape after creation, enabling
+     * dynamic resizing based on runtime requirements.
+     * <br>
+     * Note: It resets the append position to the start of the array.
+     *
+     * @param shape the new dimensions for the array
+     * @throws IllegalStateException    if the array is already closed
+     * @throws IllegalArgumentException if shape has invalid dimensions
+     */
+    public void reshape(int... shape) {
+        if (closed) {
+            throw new IllegalStateException("Cannot reshape a closed array");
+        }
+        int nDim = shape.length;
+        if (nDim > ColumnType.ARRAY_NDIMS_LIMIT) {
+            throw new IllegalArgumentException("Maximum supported dimensionality is " +
+                    ColumnType.ARRAY_NDIMS_LIMIT + "D, but got " + nDim + "D");
+        }
+        if (nDim == 0) {
+            throw new IllegalArgumentException("Shape must have at least one dimension");
+        }
+        array.setType(ColumnType.encodeArrayType(array.getElemType(), nDim));
+        for (int dim = 0; dim < nDim; dim++) {
+            array.setDimLen(dim, shape[dim]);
+        }
+        array.applyShape();
+        flatLength = array.getFlatViewLength();
+        memA = array.startMemoryA();
+    }
+
+    /**
+     * Reshapes the array to a single dimension with the specified length.
+     * <p>This is a convenience method for converting multi-dimensional arrays to
+     * one-dimensional arrays or resizing existing one-dimensional arrays without allocating an array for the shape.
+     *
+     * @param dimLen the length of the single dimension
+     * @throws IllegalStateException    if the array is already closed
+     * @throws IllegalArgumentException if dimLen is negative
+     */
+    public void reshape(int dimLen) {
+        if (closed) {
+            throw new IllegalStateException("Cannot reshape a closed array");
```

Comment:
there is also LineSenderException used elsewhere. Lets tighten up error handling please